### ISSUE TITLE
Cache getter for relations

### ISSUE THREAD

**Issue by dmachehin** [c9s/LazyRecord#118]
When i'm call in my model the relations (example: $user->projects), magic method __get create new instance of model ProjectCollection, maybe it cache?

**Comment by c9s** [comment]
Yes, there is a getRelationalRecords property: https://github.com/c9s/LazyRecord/blob/master/src/LazyRecord/BaseModel.php#L1623

I think this needs to be improved. PRs welcome :)

**Comment by c9s** [comment]
update: I think the cache mechanism should be called explicitly to avoid common pitfalls.
